app/scripts/presenters/visit_presenter.js

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'underscore'
+], function (_) {
+  'use strict';
+
+  var MINIMUM_IMAGE_ENTROPY = 1.75;
+  var MINIMUM_IMAGE_WIDTH = 270;
+  var EXTRACTED_IMAGE_DENY_PATTERN = /github.com|(\..{2,3}\/$)|([&?]q=)|([&?]page=\d+)/;
+
+  function VisitPresenter (visit) {
+    // flatten attributes
+    _.extend(this, visit.attributes, visit.get('userPage'));
+  }
+
+  _.extend(VisitPresenter.prototype, {
+    faviconUrl: function () {
+      return this.extractedFaviconUrl || '/images/icon-favicon_default@2x.png';
+    },
+
+    isSearchResult: function () {
+      return !!this.url.match(/[?&#][pq]=/i);
+    },
+
+    hasHashBang: function () {
+      return !!this.url.match(/(#!)|(#(.*?)\/)/);
+    },
+
+    hasValidExtractedImageUrl: function () {
+      return this.extractedImageUrl &&
+             (this.extractedImageEntropy && this.extractedImageEntropy > MINIMUM_IMAGE_ENTROPY) &&
+             !this.url.match(EXTRACTED_IMAGE_DENY_PATTERN);
+    },
+
+    hasLargeImage: function () {
+      return this.extractedImageWidth && this.extractedImageWidth >= MINIMUM_IMAGE_WIDTH;
+    },
+
+    getInterestingness: function () {
+      if (!this.interestingness) {
+        var value = 1.0;
+
+        // Positives
+        if (this.hasValidExtractedImageUrl()) {
+          value *= 1.3;
```

Comment:
All these multipliers are arbitrary. We'll come up with something more meaningful at some point.